### DISCUSSION TITLE
Using pyinstaller to compile python code which uses C code (as a .so)

### DISCUSSION THREAD

**Discussion by 1Codealot** [pyinstaller/pyinstaller#7663] [Help]
Hi
Is there a way to compile the my code which uses C code which uses ctypes module and I want it so that I don't need people to download the .so file?
`lib = ctypes.CDLL("URL_Creator.so")`

**Comment by bwoodsend** [comment]
Just do what it says [here in the docs](https://pyinstaller.org/en/stable/runtime-information.html?highlight=__file__#placing-data-files-at-expected-locations-inside-the-bundle) with your `.so` file but use `--add-binary` instead of `--add-data`.

**Reply by 1Codealot** [reply]
Will I also need to add
```
from os import path
path_to_dat = path.abspath(path.join(path.dirname(__file__), 'file.dat'))
```
to the python code?

**Reply by 1Codealot** [reply]
Ok I have the build work. I didn't need that, I just used this command `pyinstaller --onefile --add-binary "dotSO_File.so;." your_script.py`

**Reply by bwoodsend** [reply]
Yes but replace `file.dat` with `URL_Creator.so`. UNIX systems don't allow you to change the library search path at runtime so `ctypes.CDLL("URL_Creator.so")` needs to become equivalent to `ctypes.CDLL("/full/path/to/URL_Creator.so")` or the library will never be found. `path.abspath(path.join(path.dirname(__file__), 'URL_Creator.so'))` will find the full path to `URL_Creator.so`.
